src/contract/sent_transaction.ts

```diff
@@ -76,24 +76,27 @@ export class SentTransaction<T> {
     signTransaction: ClientOptions["signTransaction"],
     /** {@link AssembledTransaction} from which this SentTransaction was initialized */
     assembled: AssembledTransaction<U>,
+    updateTimeout: boolean = true,
   ): Promise<SentTransaction<U>> => {
     const tx = new SentTransaction(signTransaction, assembled);
-    const sent = await tx.send();
+    const sent = await tx.send({ updateTimeout });
     return sent;
   };
 
-  private send = async (): Promise<this> => {
+  private send = async ({ updateTimeout }: {updateTimeout?: boolean } =  { updateTimeout: true }): Promise<this> => {
     const timeoutInSeconds =
       this.assembled.options.timeoutInSeconds ?? DEFAULT_TIMEOUT;
-    this.assembled.built = TransactionBuilder.cloneFrom(this.assembled.built!, {
-      fee: this.assembled.built!.fee,
-      timebounds: undefined,
-      sorobanData: new SorobanDataBuilder(
-        this.assembled.simulationData.transactionData.toXDR(),
-      ).build(),
-    })
-      .setTimeout(timeoutInSeconds)
-      .build();
+    if(updateTimeout) {
+      this.assembled.built = TransactionBuilder.cloneFrom(this.assembled.built!, {
+        fee: this.assembled.built!.fee,
+        timebounds: undefined,
```

Comment:
a comment here would be helpful!
```suggestion
        timebounds: undefined, // intentionally don't clone timebounds
```